history/src/block_stores/faithful_block_store.rs

```diff
@@ -0,0 +1,54 @@
+use anyhow::bail;
+use log::warn;
+use solana_lite_rpc_core::structures::produced_block::ProducedBlock;
+use solana_rpc_client::nonblocking::rpc_client::RpcClient;
+use solana_rpc_client_api::config::RpcBlockConfig;
+use solana_sdk::clock::Slot;
+use solana_sdk::commitment_config::CommitmentConfig;
+use solana_transaction_status::{TransactionDetails, UiTransactionEncoding};
+use std::ops::RangeInclusive;
+use std::sync::Arc;
+
+pub struct FaithfulBlockStore {
+    faithful_rpc_client: Arc<RpcClient>, // to fetch legacy blocks from faithful
+}
+
+impl FaithfulBlockStore {
+    pub fn new(faithful_rpc_client: Arc<RpcClient>) -> Self {
+        Self {
+            faithful_rpc_client,
+        }
+    }
+
+    pub fn get_slot_range(&self) -> RangeInclusive<Slot> {
+        // TODO
+        warn!("slot_range not implemented for FaithfulBlockStore");
+        RangeInclusive::new(1, 0) // empty
```

Comment:
Ideally should be all the slots 0 to u64::max

Reply:
using max sounds wrong to me.
as we do not really pull in faithful I guess it's okey to go for the safe default